toolbox/fdc3-conformance/Open-Tests.md

```diff
@@ -0,0 +1,66 @@
+
+- `AOpensB1`:  **A** uses `fdc3.open(‘app B Name’)`  ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue)
+- `AOpensB2`:  **A** uses `fdc3.open({name: “<app B Name>”})` ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+- `AOpensB3`:  **A** uses `fdc3.open({name: “<app B Name>”, appId: “<app B ID>”})` ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+
+## A Fails To Open Another App
+
+| App | Step            | Description                                              |
+|-----|-----------------|----------------------------------------------------------|
+| A   | 1. Opening App  | App A tries to open a non-existent app |
+| A   | 2. Check Response | A receives an "App Not Found" error  |
+
+- `AFailsToOpenB1`:  **A** uses `fdc3.open(‘<non existent app>’)` ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue)
+- `AFailsToOpenB2`:  **A** uses `fdc3.open({name: ‘<non existent app>’})`  ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+- `AFailsToOpenB3`:  **A** uses `fdc3.open({name: ‘<non existent app>’, appId: “<non existent app id>”})`  ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+
+## A Opens B With Context
+
+| App | Step            | Description                                              |
+|-----|-----------------|----------------------------------------------------------|
+| A   | 1. Opening App     |App A opens app B with an `fdc3.instrument` context data item |
+| B   | 2. Context present | Add an untyped context listener via: <br/>`fdc3.addContextListener(null, handler)` <br /> B receives a matching piece of context data from A |
+
+- `AOpensBWithContext1`:  **A** uses `fdc3.open(‘app B Name', ctx)` ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue)
+- `AOpensBWithContext2`:  **A** uses `fdc3.open({name: “<app B Name>”}, ctx)` ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+- `AOpensBWithContext3`:  **A** uses `fdc3.open({name: “<app B Name>”, appId: “<app B ID>”}, ctx)` ![1.2](https://img.shields.io/badge/FDC3-1.2-green)
+- `AOpensBWithSpecificContext`: Perform above but replace **B**s call with `fdc3.addContextListener('fdc3.instrument', handler)` ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue)
+- `AOpensBMultipleListen`:  Perform `AOpensBWithSpecificContext` but **B** should perform an additional `fdc3.addContextListener('fdc3.contact', handler)` prior to the existing `addContextListener` for `fdc3.instrument`.  The correct context listener should receive the context, and the promise completes successfully.  ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue)
```

Comment:
Similar to previous sections, there are problems introduced by the merge of the tests...

```suggestion
## A Opens B With Context

| App | Step            | Description                                              |
|-----|-----------------|----------------------------------------------------------|
| A   | 1. Opening App     |App A opens app B with an `fdc3.instrument` Context Object by calling a function (see below) |
| B   | 2. Receive Context | Add an untyped context listener via: <br/>`fdc3.addContextListener(null, handler)` <br /> B receives an `fdc3.instrument` Context Object matching that passed to the `fdc3.open() call made by A |

- `AOpensBWithContext1`:  ![1.2](https://img.shields.io/badge/FDC3-1.2-green) **A** uses `fdc3.open(‘app B Name', <fdc3.instrument context>)` 
- `AOpensBWithContext2`:  ![1.2](https://img.shields.io/badge/FDC3-1.2-green) **A** uses `fdc3.open({name: “<app B Name>”}, <fdc3.instrument context>)` 
- `AOpensBWithContext3`:  **A** uses an `AppMetadata` or `AppIdentifier` to open B, via<br/>![1.2](https://img.shields.io/badge/FDC3-1.2-green) `fdc3.open({name: “<app B Name>”, appId: “<app B ID>”}, <fdc3.instrument context>)` <br/>![2.0](https://img.shields.io/badge/FDC3-2.0-blue)  `fdc3.open({appId: “<app B ID>”}, <fdc3.instrument context>)`
- `AOpensBWithSpecificContext`: ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue) Perform AOpensBWithContext3 but replace **B**s call with `fdc3.addContextListener('fdc3.instrument', handler)` 
- `AOpensBMultipleListen`:  ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue) Perform `AOpensBWithSpecificContext` but **B** should perform an additional `fdc3.addContextListener('fdc3.contact', handler)` prior to the existing `addContextListener` for `fdc3.instrument`.  The correct context listener should receive the context, and the promise completes successfully. 
- `AOpensBWithWrongContext`: ![1.2](https://img.shields.io/badge/FDC3-1.2-green) ![2.0](https://img.shields.io/badge/FDC3-2.0-blue) Perform `AOpensBWithSpecificContext` but **B** should add a context listener for the wrong context type (e.g. `fdc3.dummyType`) instead of the expected type in step 2<br/>Confirm that NO context is received<br/>Confirm that NO context is received <br/> ![1.2](https://img.shields.io/badge/FDC3-1.2-green) the `fdc3.open` call throws an Error with message `AppTimeout` <br/>![2.0](https://img.shields.io/badge/FDC3-2.0-blue) The promise returned to **A** by `fdc3.open` rejects with an Error with message `AppTimeout`
```